Mathlib/SetTheory/Ordinal/FixedPoint.lean

```diff
@@ -48,146 +49,138 @@ finitely many functions in the family to `a`.
 
 `Ordinal.nfpFamily_fp` shows this is a fixed point, `Ordinal.le_nfpFamily` shows it's at
 least `a`, and `Ordinal.nfpFamily_le_fp` shows this is the least ordinal with these properties. -/
-def nfpFamily (f : ι → Ordinal.{max u v} → Ordinal.{max u v}) (a : Ordinal.{max u v}) : Ordinal :=
+def nfpFamily (f : ι → Ordinal.{u} → Ordinal.{u}) (a : Ordinal.{u}) : Ordinal :=
   ⨆ i, List.foldr f a i
 
-theorem nfpFamily_eq_sup (f : ι → Ordinal.{max u v} → Ordinal.{max u v}) (a : Ordinal.{max u v}) :
-    nfpFamily.{u, v} f a = ⨆ i, List.foldr f a i :=
+@[deprecated (since := "2024-10-14")]
+theorem nfpFamily_eq_sup (f : ι → Ordinal.{u} → Ordinal.{u}) (a : Ordinal.{u}) :
+    nfpFamily f a = ⨆ i, List.foldr f a i :=
   rfl
 
-theorem foldr_le_nfpFamily (f : ι → Ordinal → Ordinal)
-    (a l) : List.foldr f a l ≤ nfpFamily.{u, v} f a :=
+theorem foldr_le_nfpFamily [Small.{u} ι] (f : ι → Ordinal.{u} → Ordinal.{u}) (a l) :
+    List.foldr f a l ≤ nfpFamily f a :=
   Ordinal.le_iSup _ _
 
-theorem le_nfpFamily (f : ι → Ordinal → Ordinal) (a) : a ≤ nfpFamily f a :=
-  Ordinal.le_iSup (fun _ ↦ List.foldr _ a _) []
+theorem le_nfpFamily [Small.{u} ι] (f : ι → Ordinal.{u} → Ordinal.{u}) (a) : a ≤ nfpFamily f a :=
+  foldr_le_nfpFamily f a []
 
-theorem lt_nfpFamily {a b} : a < nfpFamily.{u, v} f b ↔ ∃ l, a < List.foldr f b l :=
+theorem lt_nfpFamily [Small.{u} ι] {a b} : a < nfpFamily f b ↔ ∃ l, a < List.foldr f b l :=
   Ordinal.lt_iSup
 
-theorem nfpFamily_le_iff {a b} : nfpFamily.{u, v} f a ≤ b ↔ ∀ l, List.foldr f a l ≤ b :=
+theorem nfpFamily_le_iff [Small.{u} ι] {a b} : nfpFamily f a ≤ b ↔ ∀ l, List.foldr f a l ≤ b :=
   Ordinal.iSup_le_iff
 
-theorem nfpFamily_le {a b} : (∀ l, List.foldr f a l ≤ b) → nfpFamily.{u, v} f a ≤ b :=
+theorem nfpFamily_le {a b} : (∀ l, List.foldr f a l ≤ b) → nfpFamily f a ≤ b :=
   Ordinal.iSup_le
 
-theorem nfpFamily_monotone (hf : ∀ i, Monotone (f i)) : Monotone (nfpFamily.{u, v} f) := by
-  intro _ _ h
-  apply Ordinal.iSup_le
-  intro l
-  exact (List.foldr_monotone hf l h).trans (Ordinal.le_iSup _ l)
+theorem nfpFamily_monotone [Small.{u} ι] (hf : ∀ i, Monotone (f i)) : Monotone (nfpFamily f) :=
```

Comment:
If `range (List.foldr f a)` isn't small, then `nfpFamily f = 0`, right?
```suggestion
theorem nfpFamily_monotone (hf : ∀ i, Monotone (f i)) : Monotone (nfpFamily f) :=
```

Reply:
For what `a`? A counterexample of this theorem as you write it is given for `f : Ordinal → Ordinal → Ordinal` with `f x 0 = 1` and `f x a = x + a` otherwise. Then `nfpFamily f 0 = 1` but `nfpFamily f 1 = 0`.

Reply:
Hmm, okay